vnext/codegen/react/components/rnwcore/Props.h

```diff
@@ -0,0 +1,554 @@
+  bool indeterminate{false};
+  double progress{0.0};
+  bool animating{true};
+  SharedColor color{};
+  std::string testID{""};
+};
+
+enum class RCTProgressViewProgressViewStyle { Default, Bar };
+
+static inline void fromRawValue(const RawValue &value, RCTProgressViewProgressViewStyle &result) {
+  auto string = (std::string)value;
+  if (string == "default") { result = RCTProgressViewProgressViewStyle::Default; return; }
+  if (string == "bar") { result = RCTProgressViewProgressViewStyle::Bar; return; }
+  abort();
+}
+
+static inline std::string toString(const RCTProgressViewProgressViewStyle &value) {
+  switch (value) {
+    case RCTProgressViewProgressViewStyle::Default: return "default";
+    case RCTProgressViewProgressViewStyle::Bar: return "bar";
+  }
+}
+
+class RCTProgressViewProps final : public ViewProps {
+ public:
+  RCTProgressViewProps() = default;
+  RCTProgressViewProps(const RCTProgressViewProps &sourceProps, const RawProps &rawProps);
+
+#pragma mark - Props
+
+  RCTProgressViewProgressViewStyle progressViewStyle{RCTProgressViewProgressViewStyle::Default};
+  Float progress{0.0};
+  SharedColor progressTintColor{};
+  SharedColor trackTintColor{};
+  ImageSource progressImage{};
+  ImageSource trackImage{};
+};
+
+enum class AndroidSwipeRefreshLayoutSize { Default, Large };
+
+static inline void fromRawValue(const RawValue &value, AndroidSwipeRefreshLayoutSize &result) {
+  auto string = (std::string)value;
+  if (string == "default") { result = AndroidSwipeRefreshLayoutSize::Default; return; }
+  if (string == "large") { result = AndroidSwipeRefreshLayoutSize::Large; return; }
+  abort();
+}
+
+static inline std::string toString(const AndroidSwipeRefreshLayoutSize &value) {
+  switch (value) {
+    case AndroidSwipeRefreshLayoutSize::Default: return "default";
+    case AndroidSwipeRefreshLayoutSize::Large: return "large";
+  }
+}
+
+class AndroidSwipeRefreshLayoutProps final : public ViewProps {
+ public:
+  AndroidSwipeRefreshLayoutProps() = default;
+  AndroidSwipeRefreshLayoutProps(const AndroidSwipeRefreshLayoutProps &sourceProps, const RawProps &rawProps);
+
+#pragma mark - Props
+
+  bool enabled{true};
+  std::vector<SharedColor> colors{};
+  SharedColor progressBackgroundColor{};
+  AndroidSwipeRefreshLayoutSize size{AndroidSwipeRefreshLayoutSize::Default};
+  Float progressViewOffset{0.0};
+  bool refreshing{false};
+};
+
+class PullToRefreshViewProps final : public ViewProps {
+ public:
+  PullToRefreshViewProps() = default;
+  PullToRefreshViewProps(const PullToRefreshViewProps &sourceProps, const RawProps &rawProps);
+
+#pragma mark - Props
+
+  SharedColor tintColor{};
+  SharedColor titleColor{};
+  std::string title{};
+  Float progressViewOffset{0.0};
+  bool refreshing{false};
+};
+
+class SafeAreaViewProps final : public ViewProps {
+ public:
+  SafeAreaViewProps() = default;
+  SafeAreaViewProps(const SafeAreaViewProps &sourceProps, const RawProps &rawProps);
+
+#pragma mark - Props
+
```

Comment:
what are these pragmas? it doesn't look like msvc understands it?

Reply:
I wonder if this should be `#pragma region`. Google says this may be ObjC specific.

Reply:
yes, pragma region/endregion would be understood by VS/msvc